### PR TITLE
Update tree-sitter-latex and highlights

### PR DESCRIPTION
**Pull request by erasin**
update tree-sitter-latex 

update highlights from https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/latex/highlights.scm
add textobject from https://github.com/nvim-treesitter/nvim-treesitter-textobjects/blob/master/queries/latex/textobjects.scm

About highlights

I'm not sure about the markup for the Math section， It wasn't marked before. I left it as it was.

### CODE REVIEW

File: runtime/queries/latex/highlights.scm [hunk 1]
```diff
@@ -1,410 +1,236 @@
...
-  "\\textcolor"
-  "\\pagecolor"
-  "\\definecolor"
-  "\\definecolorset"
-  "\\newtheorem"
-  "\\declaretheorem"
-  "\\newacronym"
-] @function.macro
+(key_value_pair
+  key: (_) @parameter
```
Comment (the-mikedavis) [inline runtime/queries/latex/highlights.scm:9]:
```suggestion
  key: (_) @variable.parameter
```

File: runtime/queries/latex/highlights.scm [hunk 2]
```diff
@@ -1,410 +1,236 @@
...
+  command: _ @function.macro
+  from: (curly_group_text (_) @label)
+  to: (curly_group_text (_) @label))
+(label_reference
+  command: _ @function.macro
+  names: (curly_group_text_list (_) @label))
+(label_number
+  command: _ @function.macro
+  name: (curly_group_text (_) @label)
+  number: (_) @text.reference)
```
Comment (the-mikedavis) [inline runtime/queries/latex/highlights.scm:78]:
```suggestion
  number: (_) @markup.link.label)
```

I'm not really sure about this one but it seems to be the closest.

File: runtime/queries/latex/highlights.scm [hunk 3]
```diff
@@ -1,410 +1,236 @@
...
-  "\\subsection"
-  "\\subsubsection"
-  "\\paragraph"
-  "\\subparagraph"
-] @type
-
-"\\item" @punctuation.special
+  (displayed_equation)
+  (inline_formula)
+] @text.math
```
Comment (the-mikedavis) [inline runtime/queries/latex/highlights.scm:109]: Let's split these into

```scm
(displayed_equation) @markup.raw.block
(inline_formula) @markup.raw.inline
```

**Review by the-mikedavis — COMMENTED** [review]
Instead of `text.math` let's use `markup.raw`

File: runtime/queries/latex/highlights.scm (hunk 3 above)
Comment (erasin) [reply runtime/queries/latex/highlights.scm]: ok

**Review by the-mikedavis — APPROVED** [review]
Looks good, thanks for updating these!

**Pull request merged**